ignite/engine/__init__.py

```diff
@@ -499,22 +499,27 @@ def supervised_evaluation_step_amp(
 
     .. versionadded:: 0.5.0
     """
-    try:
-        from torch.cuda.amp import autocast
-    except ImportError:
-        raise ImportError("Please install torch>=1.6.0 to use amp_mode='amp'.")
 
     def evaluate_step(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
         model.eval()
         with torch.no_grad():
             x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
-            with autocast(enabled=True):
+            with _autocast(enabled=True):
                 y_pred = model(x)
             return output_transform(x, y, y_pred)
 
     return evaluate_step
 
 
+def _autocast(enabled: bool) -> Any:
```

Comment:
I don't remember where I saw this wrapper around amp but it is a good idea. Maybe to be moved in `ignite.distributed.auto.py` ?

Reply:
@sdesrozis Moved it and updated the tests. Let's wait to make sure all other tests passed.

Reply:
@sdesrozis any other comments on this PR?

Reply:
@sdesrozis what is the purpose of introducing this more or less useless function in `ignite.distributed` ?